runelite-client/src/main/java/net/runelite/client/Notifier.java

```diff
@@ -132,32 +134,53 @@ private Notifier(
 			!Strings.isNullOrEmpty(RuneLiteProperties.getLauncherVersion())
 			&& isTerminalNotifierAvailable();
 
+		flashNotification = runeliteConfig.flashNotification();
+
 		storeIcon();
 	}
 
 	public void notify(String message)
 	{
-		notify(message, TrayIcon.MessageType.NONE);
+		notify(message, TrayIcon.MessageType.NONE, null);
 	}
 
 	public void notify(String message, TrayIcon.MessageType type)
 	{
-		if (!runeLiteConfig.sendNotificationsWhenFocused() && clientUI.isFocused())
+		notify(message, type, null);
+	}
+
+	public void notify(String message, NotificationSettings customSettings)
+	{
+		notify(message, TrayIcon.MessageType.NONE, customSettings);
+	}
+
+	public void notify(String message, TrayIcon.MessageType type, NotificationSettings customSettings)
+	{
+		// if the custom settings notification state is set to "on" we should use the default RuneLite settings
+		final boolean useCustom = customSettings != null && runeLiteConfig.enableCustomNotifier() && !customSettings.notificationState().equals(NotificationSettings.States.ON);
+		final NotificationSettings config = useCustom ? customSettings : getRuneLiteNotificationSettings();
+
+		if (config.notificationState().equals(NotificationSettings.States.OFF))
 		{
 			return;
 		}
 
-		if (runeLiteConfig.requestFocusOnNotification())
+		if (!config.sendNotificationsWhenFocused() && clientUI.isFocused())
```

Comment:
This can be combined with the guard clause above as there are no assignments or operations between guards.